subprojects/core/src/main/java/org/gradle/api/internal/tasks/CacheableTaskOutputCompositeFilePropertyElementSpec.java

```diff
@@ -59,8 +59,8 @@ public OutputType getOutputType() {
     }
 
     @Override
-    public SnapshotNormalizationStrategy getSnapshotNormalizationStrategy() {
-        return parentProperty.getSnapshotNormalizationStrategy();
+    public PathNormalizationStrategy getPathNormalizationStrategy() {
+        return parentProperty.getPathNormalizationStrategy();
     }
```

Comment:
We could, but, like for `NonCacheableTaskOutputPropertySpec`, I found it better to return the normalization strategy of the parent. We cannot have a superclass for all output properties, since the types exposed to the user already extend `AbstractTaskPropertyBuilder` while the `CacheableTaskOutputCompositeFilePropertyElementSpec` doesn't. All in all, I think it is easy enough as is now. If you insist, I could replace the two calls to `parentProperty.getPathNormalizationStrategy` with returning the constant. WDYT?